workers/vita/Dockerfile

```diff
@@ -17,56 +17,77 @@ RUN sed 's/^deb \(.*\)/deb-src \1/' /etc/apt/sources.list \
-# not trivially buildable for this target so we use these packages for the
-# moment. At the time of the first build these corresponded to
-# https://github.com/vitasdk/packages/commit/b62cb68c349f8f389a4b4e57356c398c43f5531d
-# in case someone needs to rebuild them from source later
-RUN wget --progress=bar:force:noscroll -O - http://dl.vitasdk.org/openssl.tar.xz \
-	| tar --no-same-owner --no-same-permissions -C $VITASDK/arm-vita-eabi -xJf -
-RUN wget --progress=bar:force:noscroll -O - http://dl.vitasdk.org/curl.tar.xz \
-	| tar --no-same-owner --no-same-permissions -C $VITASDK/arm-vita-eabi -xJf -
-RUN wget --progress=bar:force:noscroll -O - http://dl.vitasdk.org/sdl2.tar.xz \
-	| tar --no-same-owner --no-same-permissions -C $VITASDK/arm-vita-eabi -xJf -
-RUN wget --progress=bar:force:noscroll -O - http://dl.vitasdk.org/sdl_net.tar.xz \
-	| tar --no-same-owner --no-same-permissions -C $VITASDK/arm-vita-eabi -xJf -
-
+#Install git
+RUN apt-get -y install git
+
+#Install sudo
+RUN apt-get -y install sudo
+
+#Install curl
+RUN apt-get -y install curl
+
+#Install Vita versions of most common libraries, including ogg, png, zlib etc.
+RUN git clone https://github.com/vitasdk/vdpm && \
+	cd vdpm && \
+	./install-all.sh && \
+	cd ..
+
+#Install recent SDL2 port for Vita
+RUN git clone https://github.com/rsn8887/SDL-Vita && \
+	cd SDL-Vita && \
+	make -f Makefile.vita install && \
+	cd ..
+
+#Install libmad for Vita
+RUN git clone https://github.com/Cpasjuste/libmad-psp2 && \
+	cd libmad-psp2 && \
+	make install && \
+	cd ..
+
+#Enable freetype2 on Vita by adding a patched config file
+RUN git clone https://github.com/rsn8887/vita-freetype-config && \
+	cd vita-freetype-config && \
+	chmod 777 freetype-config && \
+	cp freetype-config $VITASDK/arm-vita-eabi/bin/ && \
+	cd ..
+
+#Install faad2, theora, and mpeg
 WORKDIR /tmp/compile
 COPY common/compile-libraries.sh ${WORKER_NAME}/compile-libraries-vita.sh ./
 
-COPY common/library-rules/zlib.sh library-rules/
-RUN ./compile-libraries-vita.sh zlib
-
-COPY common/library-rules/libpng1.6.sh library-rules/
-RUN ./compile-libraries-vita.sh libpng1.6
-
-COPY common/library-rules/freetype.sh library-rules/
-RUN ./compile-libraries-vita.sh freetype
-
-COPY common/library-rules/libjpeg-turbo.sh library-rules/
-RUN ./compile-libraries-vita.sh libjpeg-turbo
-
 COPY common/library-rules/faad2.sh library-rules/
 RUN ./compile-libraries-vita.sh faad2
 
-COPY common/library-rules/libmad.sh library-rules/
-RUN ./compile-libraries-vita.sh libmad
-
-RUN ./compile-libraries-vita.sh libogg
-
 COPY common/library-rules/libtheora.sh library-rules/
 RUN ./compile-libraries-vita.sh libtheora
 
-RUN ./compile-libraries-vita.sh libvorbis
-
 COPY common/library-rules/mpeg2dec.sh library-rules/
 RUN ./compile-libraries-vita.sh mpeg2dec
 
+#Install flac support for Vita
+RUN wget --progress=bar:force:noscroll -O - \
+	http://downloads.xiph.org/releases/flac/flac-1.3.2.tar.xz \
```

Comment:
Here is again an insecure download, *and* a download which will be outdated and not receiving security updates automatically simply by rebuilding the image. Retrieving sources from the Debian repositories is again very deliberate, as these are maintained with patches for build reproducibility where needed, and with security updates, and cryptographically signed.